zqd/handlers_test.go

```diff
@@ -315,6 +319,65 @@ func TestPostInvalidLogs(t *testing.T) {
 	})
 }
 
+func TestDeleteDuringPacketPost(t *testing.T) {
+	c, client, done := newCore(t)
+	defer done()
+
+	spaceName := "deleteDuringPacketPost"
+	pcapfile := "./testdata/valid.pcap"
+
+	_, err := client.SpacePost(context.Background(), api.SpacePostRequest{Name: spaceName})
+	require.NoError(t, err)
+
+	waitFn := func(tzp *testZeekProcess) error {
+		select {
+		case <-tzp.ctx.Done():
+		}
+		return tzp.ctx.Err()
+	}
+
+	c.ZeekLauncher = testZeekLauncher(nil, waitFn)
+
+	var wg sync.WaitGroup
+	packetPostErr := make(chan error)
+
+	wg.Add(1)
+	doPost := func() error {
+		stream, err := client.PacketPost(context.Background(), spaceName, api.PacketPostRequest{pcapfile})
+		if err != nil {
+			return err
+		}
+		wg.Done()
+
+		var taskEnd *api.TaskEnd
+		for {
+			i, err := stream.Next()
+			if err != nil {
+				return err
+			}
+			if i == nil {
+				break
+			}
+			if te, ok := i.(*api.TaskEnd); ok {
+				taskEnd = te
+			}
+		}
+		if taskEnd == nil || taskEnd.Error.Message != "context canceled" {
+			return errors.New("missing or unexpected taskEnd error")
+		}
+		return nil
+	}
+	go func() {
+		packetPostErr <- doPost()
+	}()
+
+	wg.Wait()
+	err = client.SpaceDelete(context.Background(), spaceName)
+	require.NoError(t, err)
+
+	require.NoError(t, <-packetPostErr)
```

Comment:
I think something like this would be make this test more clear:
```suggestion
		if taskEnd != nil  {
			return taskEnd.Error
		}
		return nil
	}
	go func() {
		packetPostErr <- doPost()
	}()

	wg.Wait()
	err = client.SpaceDelete(context.Background(), spaceName)
	require.NoError(t, err)

	require.Error(t, <-packetPostErr, "context canceled")
```